modules/base/tool.go

```diff
@@ -206,11 +206,26 @@ func HashEmail(email string) string {
 
 // AvatarLink returns avatar link by given email.
 func AvatarLink(email string) string {
-	if setting.DisableGravatar || setting.OfflineMode {
-		return setting.AppSubUrl + "/img/avatar_default.png"
+	var url string
+
+	if !setting.OfflineMode {
+		if setting.EnableFederatedAvatar && setting.LibravatarService != nil {
```

Comment:
I will do the fix if the code actually works, but just coding style issues. You can leave it to me.